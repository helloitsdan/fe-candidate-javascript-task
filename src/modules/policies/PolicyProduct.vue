<template>
  <div class="text-center">
    <h2>{{ type }} product</h2>

    <div class="columns">
      <div class="column">
        <span class="h1 display-block">{{ price.monthly | formatCurrency }}</span>
        monthly
      </div>
      <div class="column">
        <span class="h1 display-block">{{ price.annual | formatCurrency }}</span>
        annually
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "PolicyProduct",
  props: {
    type: String,
    price: Object
  }
});
</script>
