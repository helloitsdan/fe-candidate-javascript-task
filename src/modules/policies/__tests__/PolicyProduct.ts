import { shallowMount } from "@vue/test-utils";

import PolicyProduct from "../PolicyProduct.vue";

describe("PolicyProduct", () => {
  const fakeEntity = {
    type: "BASIC",
    price: {
      monthly: 100.0,
      annual: 1200.0,
    }
  };

  it("renders", () => {
    const wrapper = shallowMount(PolicyProduct, {
      propsData: fakeEntity
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
