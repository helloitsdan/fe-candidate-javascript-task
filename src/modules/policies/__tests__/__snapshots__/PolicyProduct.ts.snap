// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyProduct renders 1`] = `
<div
  class="text-center"
>
  <h2>
    BASIC product
  </h2>
   
  <div
    class="columns"
  >
    <div
      class="column"
    >
      <span
        class="h1 display-block"
      >
        £100.00
      </span>
      
      monthly
    
    </div>
     
    <div
      class="column"
    >
      <span
        class="h1 display-block"
      >
        £1200.00
      </span>
      
      annually
    
    </div>
  </div>
</div>
`;
